contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -543,6 +552,27 @@ add_rewritten_query_fragment_to_mutator(PLtsql_expr_query_mutator *mutator)
 		mutator->add(entry.first, entry.second.first, entry.second.second);
 }
 
+static void
+add_query_hints(PLtsql_expr *expr)
+{
+	std::string hint =  "/*+ ";
+	for(auto q_hint: query_hints) {
+		hint += q_hint;
+		hint += " ";
+	}
+	hint += "*/";
+	StringInfoData new_query;
+	initStringInfo(&new_query);
+	appendStringInfo(&new_query, "%s %s", const_cast <char *>(hint.c_str()), expr->query);
```

Comment:
nit: why do we need `const_cast`? We have to avoid using const_cast as much as possible. I think `construct_unique_index_name` still needs `char *` (not `const char*`)  so we have to keep it but other function such as `extractIndexValues()` should use const char*.